### PR TITLE
Support repository_id in org ruleset conditions

### PR DESCRIPTION
**Pull request by ysoldak**
This PR adds support for referencing repositories by ID in Ruleset conditions.
See https://docs.github.com/en/rest/orgs/rules?apiVersion=2022-11-28#create-an-organization-repository-ruleset

API is changed a bit, probably a low impact since v53.2.0 (with ruleset support added) was released just recently and hopefully not many have code using it in production.

The change `RulesetRepositoryConditionParameters` -> `RulesetRepositoryNamesConditionParameters`

I'm open for suggestions for a change that would not shake the API.

### CODE REVIEW

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:50]:
```suggestion
// Set either RepositoryName or RepositoryID, not both.
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:46]:
```suggestion
// RulesetRepositoryIDsConditionParameters represents the conditions object for repository_ids.
type RulesetRepositoryIDsConditionParameters struct {
	RepositoryIDs []int64 `json:"repository_ids,omitempty"`
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:54]:
```suggestion
	RepositoryID   *RulesetRepositoryIDsConditionParameters   `json:"repository_id,omitempty"`
```

File: github/orgs_rules_test.go
Comment (gmlewis) [inline github/orgs_rules_test.go:74]:
```suggestion
func TestOrganizationsService_CreateOrganizationRuleset_RepoNames(t *testing.T) {
```

File: github/orgs_rules_test.go
Comment (gmlewis) [inline github/orgs_rules_test.go:395]:
```suggestion
func TestOrganizationsService_CreateOrganizationRuleset_RepoIDs(t *testing.T) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @ysoldak !
Just a few tweaks, please, then we will be ready for a second LGTM+Approval before merging.

**Comment by gmlewis** [comment]
Please also update the unit tests so that `go test ./...` passes on your local machine.

**Comment by ysoldak** [comment]
🤦 PTAL

**Review by gmlewis — APPROVED** [review]
Thank you, @ysoldak !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by ysoldak** [comment]
Thanks, @gmlewis

I thought worth to reference GitHub's own statement about API change:
https://github.blog/changelog/2023-06-27-repository-rules-public-beta-updates/

The part this PR addresses is "Added a repository picker to target select repositories for organization rulesets."

**Review by liaodaniel — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @liaodaniel !
Merging.

**Pull request merged**
